cmd/operator/main.go

```diff
@@ -393,6 +393,35 @@ func run(fs *flag.FlagSet) int {
 		}
 	}
 
+	// If Prometheus Agent runs in DaemonSet mode, check if
+	// the operator has proper RBAC permissions on the DaemonSet resource.
+	if cfg.Gates.Enabled(operator.PrometheusAgentDaemonSetFeature) {
+		daemonSetGroupVersion := schema.GroupVersion{
+			Group:   "apps",
+			Version: "v1",
+		}
+
+		allowed, errs, err := k8sutil.IsAllowed(ctx,
+			kclient.AuthorizationV1().SelfSubjectAccessReviews(),
+			cfg.Namespaces.PrometheusAllowList.Slice(),
+			k8sutil.ResourceAttribute{
+				Group:    daemonSetGroupVersion.Group,
+				Version:  daemonSetGroupVersion.Version,
+				Resource: "daemonset",
+				Verbs:    []string{"get", "list", "watch"},
+			})
+		if err != nil {
+			level.Error(logger).Log("msg", "failed to check permissions on DaemonSet resource", "err", err)
+			cancel()
+			return 1
+		}
+		if !allowed {
+			for _, reason := range errs {
+				level.Error(logger).Log("msg", "missing permissions to manage Daemonset resource for Prometheus Agent", "reason", reason)
```

Comment:
Either we exit if the permissions aren't sufficient or we pass the information to the prometheus agent controller. I'd go with option 1 for now (if you enable the feature gate, you expect the controller to be functional).

Reply:
+1 for exiting

Reply:
yeah let's exit for now